pkg/config/identity/api_key.go

```diff
@@ -1,13 +1,88 @@
 package identity
 
 import (
+	"context"
+	"fmt"
 	"github.com/3scale-labs/authorino/pkg/config/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 type APIKey struct {
-	SecretKey string `yaml:"secret_key"`
+	Name           string            `yaml:"name"`
+	Location       string            `yaml:"location"`
+	KeySelector    string            `yaml:"key_selector"`
+	LabelSelectors map[string]string `yaml:"label_selectors"`
 }
 
-func (self *APIKey) Call(ctx common.AuthContext) (interface{}, error) {
-	return "Authenticated with API key", nil // TODO: implement
+const (
+	authorinoNamespace = "authorino"
+)
+
+var (
+	clientSet *kubernetes.Clientset
+	apiKeyLog = ctrl.Log.WithName("Authorino").WithName("ApiKey")
+)
+
+func setupClient() (clientSet *kubernetes.Clientset) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		apiKeyLog.Error(err, "something went terribly wrong getting the config")
+		panic(err.Error())
+	}
+
+	clientSet, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		apiKeyLog.Error(err, "something went terribly wrong setting the clientSet")
+		panic(err.Error())
+	}
+	return
+}
+
+func init() {
+	clientSet = setupClient()
+}
+
+func (apiKey *APIKey) Call(authCtx common.AuthContext) (interface{}, error) {
+	labelSelectors := metav1.LabelSelector{MatchLabels: apiKey.LabelSelectors}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelectors.MatchLabels).String(),
+	}
+
+	secrets, err :=
+		clientSet.CoreV1().Secrets(authorinoNamespace).List(
+			context.TODO(),
+			listOptions)
+	if err != nil {
+		apiKeyLog.Error(err, "something went terribly wrong getting the secrets")
+	}
+
+	reqHeaders := authCtx.GetRequest().Attributes.Request.Http.Headers
+	fmt.Println("Le reqHeaders", reqHeaders)
+	reqPath := authCtx.GetRequest().GetAttributes().GetRequest().GetHttp().GetPath()
+	fmt.Println("Le reqPath", reqPath)
+	reqQuery := authCtx.GetRequest().GetAttributes().GetRequest().GetHttp().GetQuery()
+	fmt.Println("Le reqQuery", reqQuery)
+	reqFragment := authCtx.GetRequest().GetAttributes().GetRequest().GetHttp().GetFragment()
+	fmt.Println("Le reqFragment", reqFragment)
```

Comment:
I think we'll be using the `Location` field for this (probably in a `switch..case` statement), right?

Reply:
Yup. Also, don't trouble yourself with review just now since it's super WIP hehe